test/regression/runSingleTest.ts

```diff
@@ -3,22 +3,83 @@ import {By, snap, Snappit, IConfig, WebDriver, WebElement} from "snappit-visual-
 
 import * as util from "../common/util";
 
-export async function regressionTest(browser: string, component: string) {
-    
+
+export async function regressionTest(browser: string, component: string, buildIdentifier: string) {
+    const runningSauce = process.env.SAUCE_USERNAME && process.env.SAUCE_ACCESS_KEY;
+    interface IResolutions {
+        [key:string]: string
+        firefox: string;
+        chrome: string;
+        safari: string;
+        "internet explorer": string;
+        MicrosoftEdge: string;
+    }
+
+    const screenResolution = "1600x1200";
+
+    interface IPlatforms {
+        [key: string]: string
+        firefox: string;
+        chrome: string;
+        safari: string;
+        "internet explorer": string;
+        MicrosoftEdge: string;
+    }
+
+    const MAC = "macOS 10.13";
+    const platforms: IPlatforms = {
+        firefox: "Windows 10",
+        chrome: MAC,
+        safari: MAC,
+        "internet explorer": "Windows 10",
+        MicrosoftEdge: "Windows 10"
+    };
+    const platform = platforms[browser];
+
     const config: IConfig = {
-        browser: browser
+        browser: browser,
+        logException: [
+            "MISMATCH",
+            "NO_BASELINE",
+            "SIZE_DIFFERENCE",
+        ],
+        sauceLabs: {
+            platform: platform,
+            version: "latest",
+            screenResolution: screenResolution,
+            extendedDebugging: true,
+            maxDuration: 120,
+        }
     };
 
-    test(`${browser} -- ${component}`, async t => {
+    test(`${browser} on ${platform} -- ${component}`, async t => {
+        
+        if (runningSauce) {
+            config.serverUrl = `http://${process.env.SAUCE_USERNAME}:${process.env.SAUCE_ACCESS_KEY}@ondemand.saucelabs.com:80/wd/hub`;
```

Comment:
What happens if you change this to `https://`?

I tried to find an example online of someone using https for this, but every single one used http.